From 4325be4d7f68f441dc7cc49449e8fd8f40f11757 Mon Sep 17 00:00:00 2001
From: Xabier Oneca <xoneca@gmail.com>
Date: Fri, 28 Sep 2018 13:49:54 +0200
Subject: [PATCH] fix compilation with --enable-redir and musl

musl lacks `allocated` member on `struct re_pattern_buffer` (`regex_t`)
(https://fossies.org/dox/musl-1.1.20/structre__pattern__buffer.html).

Patch inspired by drkhosla's comment at issue #200
(https://github.com/coova/coova-chilli/issues/200#issuecomment-253497070)
---
 src/main-redir.c | 21 +++++++--------------
 src/options.c    |  2 +-
 2 files changed, 8 insertions(+), 15 deletions(-)

From 6aaf47c1cf0a9c27743e39b0fc02dacb89e46cd0 Mon Sep 17 00:00:00 2001
From: esaaprillia <esaapriliasalsabila@gmail.com>
Date: Tue, 22 Apr 2025 21:16:05 +0800
Subject: [PATCH] Update main-redir.c

---
 src/main-redir.c | 19 ++++++-------------
 1 file changed, 6 insertions(+), 13 deletions(-)

diff --git a/src/main-redir.c b/src/main-redir.c
index 2591113..d6c8fe2 100644
--- a/src/main-redir.c
+++ b/src/main-redir.c
@@ -510,19 +510,12 @@ check_regex(regex_t *re, char *regex, char *s) {
   log_dbg("Checking %s =~ %s", s, regex);
 #endif
 
-#if defined (__FreeBSD__) || defined (__APPLE__) || defined (__OpenBSD__) || defined (__NetBSD__)
-  if (!re->re_g)
-#else
-  if (!re->allocated) 
-#endif
-  {
-    if ((ret = regcomp(re, regex, REG_EXTENDED | REG_NOSUB)) != 0) {
-      char error[512];
-      regerror(ret, re, error, sizeof(error));
-      log_err(0, "regcomp(%s) failed (%s)", regex, error);
-      regex[0] = 0;
-      return -1;
-    }
+  if ((ret = regcomp(re, regex, REG_EXTENDED | REG_NOSUB)) != 0) {
+    char error[512];
+    regerror(ret, re, error, sizeof(error));
+    log_err(0, "regcomp(%s) failed (%s)", regex, error);
+    regex[0] = 0;
+    return -1;
   }

   if ((ret = regexec(re, s, 0, 0, 0)) == 0) {
 
diff --git a/src/options.c b/src/options.c
index eef2171c..7dc6f5c6 100644
--- a/src/options.c
+++ b/src/options.c
@@ -377,7 +377,7 @@ int options_fromfd(int fd, bstring bt) {
 
 #ifdef ENABLE_CHILLIREDIR
   for (i = 0; i < MAX_REGEX_PASS_THROUGHS; i++) {
-#if defined (__FreeBSD__) || defined (__APPLE__) || defined (__OpenBSD__) || defined (__NetBSD__)
+#if defined (__linux__) || defined (__FreeBSD__) || defined (__APPLE__) || defined (__OpenBSD__) || defined (__NetBSD__)
     regfree(&_options.regex_pass_throughs[i].re_host);
     regfree(&_options.regex_pass_throughs[i].re_path);
     regfree(&_options.regex_pass_throughs[i].re_qs);
